### ISSUE TITLE
Make `ToBuiltin` handle any kind of built-in list

### ISSUE THREAD

**Issue by effectfully** [IntersectMBO/plutus#5924]
`PlutusTx.Builtins.Class` has the following:

```haskell
instance ToBuiltin [BuiltinData] (BuiltinList BuiltinData) where
    {-# INLINABLE toBuiltin #-}
    toBuiltin []     = mkNilData unitval
    toBuiltin (d:ds) = mkCons d (toBuiltin ds)

instance ToBuiltin [(BuiltinData, BuiltinData)] (BuiltinList (BuiltinPair BuiltinData BuiltinData)) where
    {-# INLINABLE toBuiltin #-}
    toBuiltin []     = mkNilPairData unitval
    toBuiltin (d:ds) = mkCons (toBuiltin d) (toBuiltin ds)
```

We should be able to handle any kind of built-in list, since we don't really need those `mkNilData` and `mkNilPairData` builtins as we can simply embed the empty list constants directly.

**Comment by effectfully** [comment]
`ToBuiltin` is now called `ToOpaque`.

**Comment by effectfully** [comment]
Was fixed by #6347.
